### PR TITLE
Fixing bug #150 (Implicit namespaces in actions)

### PR DESCRIPTION
**Pull request by aaronjeline**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy-validator/src/lib.rs [hunk 1]
```diff
@@ -319,4 +319,71 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_action_namespace_inference_multi() {
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/lib.rs:324]: I'd rather move this test to `schema.rs`, and then update it to specifically assert that schema that's built contains an action with the expected `memberOf`. Also the schema can use `json!` macro.

File: cedar-policy-validator/src/schema.rs [hunk 2]
```diff
@@ -1013,7 +1019,7 @@ impl ValidatorSchema {
             for p_entity in action.applies_to.applicable_principal_types() {
                 match p_entity {
                     EntityType::Concrete(p_entity) => {
-                        if !entity_types.contains_key(p_entity) {
+                        if !entity_types.contains_key(&p_entity) {
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/schema.rs:1022]: No `&` here (and below). vscode's rust-analyzer and clippy disagree on this, and this line has churned a couple times already

**Review by john-h-kastner-aws — CHANGES_REQUESTED** [review]

File: cedar-policy-validator/src/schema.rs [hunk 3]
```diff
@@ -621,7 +621,12 @@ impl ValidatorNamespaceDef {
         namespace: Option<&Name>,
     ) -> Result<EntityUID> {
         let namespaced_action_type = if let Some(action_ty) = &action_id.ty {
-            Name::from_normalized_str(action_ty).map_err(SchemaError::EntityTypeParseError)?
+            let name =
```
[the remark this answers is not in the document]
Comment (aaronjeline) [reply cedar-policy-validator/src/schema.rs:624]: That is different behavior than I implemented, but we want to go with that behavior I assume?

[the remark this answers is not in the document]
Comment (john-h-kastner-aws) [reply cedar-policy-validator/src/schema.rs:624]: Yes. Entity types should be qualified according to the same rules everywhere

File: cedar-policy-core/src/ast/name.rs [hunk 4]
```diff
@@ -304,6 +304,8 @@ impl<'a> arbitrary::Arbitrary<'a> for Id {
 
 #[cfg(test)]
 mod test {
+    use std::str::FromStr;
```
Comment (john-h-kastner-aws) [inline cedar-policy-core/src/ast/name.rs:307]: nit: leftover import

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-policy-core/src/ast/name.rs (hunk 4 above)
Comment (john-h-kastner-aws) [reply cedar-policy-core/src/ast/name.rs:307]:
```suggestion
```

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-policy-validator/src/schema.rs [hunk 5]
```diff
@@ -621,7 +621,8 @@ impl ValidatorNamespaceDef {
         namespace: Option<&Name>,
     ) -> Result<EntityUID> {
         let namespaced_action_type = if let Some(action_ty) = &action_id.ty {
-            Name::from_normalized_str(action_ty).map_err(SchemaError::EntityTypeParseError)?
+            Self::parse_possibly_qualified_name_with_default_namespace(action_ty, namespace)
+                .map_err(SchemaError::NamespaceParseError)?
```
Comment (cdisselkoen) [inline cedar-policy-validator/src/schema.rs:625]: Error type changed from `EntityTypeParseError` to `NamespaceParseError`, was that intentional?

**Review by cdisselkoen — APPROVED** [review]

File: cedar-policy-validator/src/schema.rs (hunk 5 above)
Comment (john-h-kastner-aws) [reply cedar-policy-validator/src/schema.rs:625]: Good catch. That does not look right.

Comment (aaronjeline) [reply cedar-policy-validator/src/schema.rs:625]: Sure, I wasn't sure which one we wanted, since we are parsing a namespace as well?

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**